vivarium/examples/disease_model/population.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+import pandas as pd
+
+
+class NonCRNBasePopulation:
+
+    configuration_defaults = {
+        'population': {
+            'age_start': 0,
+            'age_end': 100,
+        }
+    }
+
+    def setup(self, builder):
+        self.config = builder.configuration
+        self.randomness = builder.randomness.get_stream('population_age_fuzz')
+
+        columns_created = ['age', 'sex', 'alive', 'entrance_time']
+        builder.population.initializes_simulants(self.generate_test_population, creates_columns=columns_created)
+
+        self.population_view = builder.population.get_view(columns_created)
+
+        builder.event.register_listener('time_step', self.age_simulants)
+
+    def generate_test_population(self, pop_data):
+        age_start = pop_data.user_data.get('age_start', self.config.population.age_start)
+        age_end = pop_data.user_data.get('age_end', self.config.population.age_end)
+
+        if age_start == age_end:
+            age = (age_start
+                   + self.randomness.get_draw(pop_data.index) * (pop_data.creation_window / pd.Timedelta(days=365)))
+        else:
+            age = age_start + self.randomness.get_draw(pop_data.index) * (age_end - age_start)
+
+        population = pd.DataFrame(
+            {'age': age,
+             'sex': self.randomness.choice(pop_data.index, ['Male', 'Female'], additional_key='sex_choice'),
+             'alive': pd.Series('alive', index=pop_data.index),
+             'entrance_time': pop_data.creation_time},
+            index=pop_data.index)
+
+        self.population_view.update(population)
+
+    def age_simulants(self, event):
+        population = self.population_view.get(event.index, query="alive == 'alive'")
+        population['age'] += event.step_size / pd.Timedelta(days=365)
+        self.population_view.update(population)
+
+
+class BasePopulation(NonCRNBasePopulation):
```

Comment:
Why have two classes and inheritance here?  Would one class be sufficient?

Reply:
They're both here so I can write a demo about the effects of CRN on variance in the system.  Do you think this is the wrong place for that?

Reply:
Actually, I have an idea for this.  Update forthcoming.